File: NativeTest/app/src/main/java/com/mcm/nativetest/poseestimation/TargetConstants.java
package com.mcm.nativetest.poseestimation;

import edu.wpi.first.wpilibj.geometry.Pose2d;
import edu.wpi.first.wpiutil.math.Pair;

import java.util.HashMap;

public final class TargetConstants {

    public static Pose2d getMarkerPose(int code) {
        return targetMap.get(code);
    }

    private static final HashMap<Integer, Pose2d> targetMap = createMap(
            Pair.of(1, new Pose2d())
    );

    private static HashMap<Integer, Pose2d> createMap(Pair<Integer, Pose2d>... pairs) {
        HashMap<Integer, Pose2d> map = new HashMap<>();
        for(Pair<Integer, Pose2d> pair: pairs) map.put(pair.getFirst(), pair.getSecond());
        return map;
    }
}

File: NativeTest/app/src/main/java/com/mcm/nativetest/poseestimation/TargetConstants.kt
package com.mcm.nativetest.poseestimation

import edu.wpi.first.wpilibj.geometry.Pose2d
import com.mcm.nativetest.poseestimation.TargetConstants
import edu.wpi.first.wpilibj.geometry.Rotation2d
import edu.wpi.first.wpiutil.math.Pair
import java.util.HashMap

private val Number.degrees get() = Rotation2d.fromDegrees(toDouble())

object TargetConstants {
    @JvmStatic
    fun getMarkerPose(code: Int): Pose2d? {
        return targetMap[code]
    }

    private val targetMap = hashMapOf(
        1 to Pose2d(0.0, 0.06, 0.degrees),
        2 to Pose2d(0.0, -0.06, 0.degrees)
    )
}